server/resp/src/main/java/org/infinispan/server/resp/commands/list/LPOS.java

```diff
@@ -53,8 +52,7 @@ public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
       return handler.stageToReturn(cs, ctx, (res, buff) -> {
          if (res == handler) return;
 
-         if (res == null || res instanceof Long) Consumers.LONG_BICONSUMER.accept((Long) res, buff);
-         else Consumers.COLLECTION_LONG_BICONSUMER.accept((Collection<Long>) res, buff);
+         Resp3Response.serialize(res, buff);
```

Comment:
Looks like this command can only return a Collection<Long> or a single Long. We could add an if check and call to the appropriate serializer then. Another option is you could add a record that encapsulates the value and the serializer in the lposAndReturn method.